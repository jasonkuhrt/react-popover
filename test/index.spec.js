import React, { PropTypes as T } from "react"
import Renderer from "react-test-renderer"
import portalMixin from "../lib/react-layer-mixin"



class ContextProvider extends React.Component {
  static propTypes = {
    children: T.node,
  }
  static defaultProps = {
    children: null,
  }
  static childContextTypes = {
    x: T.string,
  }
  getChildContext () {
    return ({
      x: "value-from-context",
    })
  }
  render () {
    return (
      this.props.children
    )
  }
}



class ContextDependent extends React.Component {
  static contextTypes = {
    x: T.string,
  }
  render () {
    return (
      <span>{this.context.x}</span>
    )
  }
}

const Portal = React.createClass({
  propTypes: {
    children: T.node
  },
  mixins: [portalMixin()],
  renderLayer () {
    return (
      this.props.children
    )
  },
  render () {
    return null
  }
})



beforeEach(() => {
  document.body.innerHTML = ""
})

it("Pass context through to layer", () => {
  Renderer.create(
    <ContextProvider>
      <Portal>
        <ContextDependent />
      </Portal>
    </ContextProvider>
  )
  expect(document.body.innerHTML).toMatchSnapshot()
})
