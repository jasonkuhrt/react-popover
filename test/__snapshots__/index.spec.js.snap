// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pass context through to layer 1`] = `"<div><span data-reactroot=\\"\\">value-from-context</span></div>"`;
